src/package-linker.js

```diff
@@ -449,8 +449,8 @@ export default class PackageLinker {
       const range = peerDeps[name];
       const pkgs = this.resolver.getAllInfoForPackageName(name);
       const found = pkgs.find(pkg => {
-        const {root, version} = pkg._reference || {};
-        return root && this._satisfiesPeerDependency(range, version);
+        const {level, version} = pkg._reference || {};
+        return level <= ref.level && this._satisfiesPeerDependency(range, version);
```

Comment:
I think we should be prioritizing some links over others, i.e. top level dependency should satisfy its dependencies from root (same level). I'm not sure whether a nested dependency should satisfy it from root first or sibling first. But maybe instead of find, we need to add it to a collection and then pick out either the lowest level or closest level match. WDYT?

Reply:
Good thinking! I think it should resolve to the closest. I'll update the code.